drivers/adc/adc_renesas_ra.c

```diff
@@ -386,4 +459,10 @@ static int adc_ra_init(const struct device *dev)
 	DEVICE_DT_INST_DEFINE(idx, adc_ra_init, NULL, &adc_ra_data_##idx, &adc_ra_config_##idx,    \
 			      POST_KERNEL, CONFIG_ADC_INIT_PRIORITY, &adc_ra_api_##idx)
 
-DT_INST_FOREACH_STATUS_OKAY(ADC_RA_INIT);
+#define DT_DRV_COMPAT renesas_ra_adc12
+DT_INST_FOREACH_STATUS_OKAY_VARGS(ADC_RA_INIT_VARIANT, ADC_VARIANT_ADC12, 12, ADC_RESOLUTION_12_BIT)
+#undef DT_DRV_COMPAT
+
+#define DT_DRV_COMPAT renesas_ra_adc16
+DT_INST_FOREACH_STATUS_OKAY_VARGS(ADC_RA_INIT_VARIANT, ADC_VARIANT_ADC16, 16, ADC_RESOLUTION_16_BIT)
+#undef DT_DRV_COMPAT
```

Comment:
It's mostly a matter of preference, but using `DT_FOREACH_STATUS_OKAY`
may be easier to understand since it doesn't require `#endif`.

Like as
https://github.com/zephyrproject-rtos/zephyr/blob/5805685c326b8d7b6385c4ad910b22c9c762adcb/drivers/display/ssd1306.c#L575-L577